api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -59,26 +55,92 @@ public class FhirObservationDaoImpl extends BaseFhirDao<Obs> implements FhirObse
 	private FhirEncounterDao encounterDao;
 	
 	@Override
-	public List<Integer> getSearchResultIds(@Nonnull SearchParameterMap theParams) {
+	public List<Obs> getSearchResults(@Nonnull SearchParameterMap theParams) {
 		if (!theParams.getParameters(FhirConstants.LASTN_OBSERVATION_SEARCH_HANDLER).isEmpty()) {
 			Criteria criteria = getSessionFactory().getCurrentSession().createCriteria(typeToken.getRawType());
 			
 			setupSearchParams(criteria, theParams);
 			
-			criteria.setProjection(
-			    Projections.projectionList().add(property("id")).add(property("concept")).add(property("obsDatetime")));
+			criteria.addOrder(Order.asc("concept")).addOrder(Order.desc("obsDatetime"));
 			
-			@SuppressWarnings("unchecked")
-			List<LastnResult<Integer>> results = ((List<Object[]>) criteria.list()).stream().map(objects -> {
-				Map<String, Object> attributes = new HashMap<>();
-				attributes.put("concept", objects[1]);
-				return new LastnResult<>((Integer) objects[0], (Date) objects[2], attributes);
-			}).collect(Collectors.toList());
+			List<Obs> results = new ArrayList<>();
+			int firstResult = 0;
+			int maxGroupCount = getMaxParameter(theParams);
+			final int BATCH_SIZE = 3;
```

Comment:
I'm not sure I quite understand why we're using the constant 3 here?

Reply:
I was testing the algorithm... I forgot to revert it back to 100. Good eye ;)